cibuildwheel/platforms/android.py

```diff
@@ -0,0 +1,625 @@
+                set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+                """
+            ),
+            file=toolchain_file,
+        )
+
+
+def localize_sysconfigdata(
+    python_dir: Path, build_env: dict[str, str], sysconfigdata_path: Path
+) -> dict[str, Any]:
+    sysconfigdata: dict[str, Any] = run_path(str(sysconfigdata_path))["build_time_vars"]
+    with sysconfigdata_path.open("w") as f:
+        f.write("# Generated by cibuildwheel\n")
+        f.write("build_time_vars = ")
+        sysconfigdata = localized_vars(build_env, sysconfigdata, python_dir / "prefix")
+        pprint(sysconfigdata, stream=f, compact=True)
+        return sysconfigdata
+
+
+def localized_vars(
+    build_env: dict[str, str], orig_vars: dict[str, Any], prefix: Path
+) -> dict[str, Any]:
+    orig_prefix = orig_vars["prefix"]
+    localized_vars_ = {}
+    for key, value in orig_vars.items():
+        # The host's sysconfigdata will include references to build-time paths.
+        # Update these to refer to the current prefix.
+        final = value
+        if isinstance(final, str):
+            final = final.replace(orig_prefix, str(prefix))
+
+        if key == "ANDROID_API_LEVEL":
+            if api_level := build_env.get(key):
+                final = int(api_level)
+
+        # Build systems vary in whether FLAGS variables are read from sysconfig, and if so,
+        # whether they're replaced by environment variables or combined with them. Even
+        # setuptools has changed its behavior here
+        # (https://github.com/pypa/setuptools/issues/4836).
+        #
+        # Ensure consistency by clearing the sysconfig variables and letting the environment
+        # variables take effect alone. This will also work for any non-Python build systems
+        # which the build script may call.
+        elif key in ["CFLAGS", "CXXFLAGS", "LDFLAGS"]:
+            final = ""
+
+        # These variables contain an embedded copy of LDFLAGS.
+        elif key in ["LDSHARED", "LDCXXSHARED"]:
+            final = final.removesuffix(" " + orig_vars["LDFLAGS"])
+
+        localized_vars_[key] = final
+
+    return localized_vars_
+
+
+def setup_android_env(
+    python_dir: Path, venv_dir: Path, build_env: dict[str, str]
+) -> dict[str, str]:
+    site_packages = next(venv_dir.glob("lib/python*/site-packages"))
+    for suffix in ["pth", "py"]:
+        shutil.copy(resources.PATH / f"_cross_venv.{suffix}", site_packages)
+
+    sysconfigdata_path = Path(
+        shutil.copy(
+            next(python_dir.glob("prefix/lib/python*/_sysconfigdata_*.py")),
+            site_packages,
+        )
+    )
+    sysconfigdata = localize_sysconfigdata(python_dir, build_env, sysconfigdata_path)
+
+    android_env = build_env.copy()
+    android_env["CIBW_CROSS_VENV"] = "1"  # Activates the code in _cross_venv.py.
+
+    env_output = call(python_dir / "android.py", "env", env=build_env, capture_stdout=True)
+    for line in env_output.splitlines():
```

Comment:
Can we be sure that environment variables won't contain newlines? We've seen a few occasions where that wasn't the case. Generally I prefer to use JSON as the format to pass env vars between processes like this, if possible.

Reply:
They certainly won't contain newlines at the moment, but I've updated the code to be able to handle them.